### PR TITLE
add check command and replace optimist with minimist

### LINKED ISSUE #292: add command to check style
I was initially thinking on keeping this project as simple as possible and let other tools *derive* from it, but I think we should have our own `check` command (always be in sync with latest release). eg:

```sh
# check if file is formatted properly
esformatter check foo.js
```

ideally do something similar to what we do during the unit tests (highlight *expected* tokens as green and *actual* tokens as red) but displaying only the lines that have differences.

we should also add an option for [unified diff](http://en.wikipedia.org/wiki/Diff_utility#Unified_format) (with and without color) since other tools can consume this format and some terminals don't support color.

```sh
# unified diff without color
esformatter check --unified-no-color foo.js
```

the [diff](https://www.npmjs.com/package/diff) module is a good option.

### PR DESCRIPTION
**Pull request by millermedeiros**
this is what the default output looks like:

screen shot 2015-04-01 at 16 18 06

`--unified` flag:

screen shot 2015-04-01 at 16 19 27

and `--unified-no-color`:

screen shot 2015-04-01 at 16 20 40

---

I did not write any tests for the `check` command on the cli because I'm planning to actually rewrite all the cli tests in the near future (see #296)

also simplified the way errors are logged

closes #292

### CODE REVIEW

**Comment by millermedeiros** [comment]
cool thing is that for large outputs we can pipe into `less(1)`: `esformatter check js/*.js | less -r` so we can move up <kbd>u</kbd>, down <kbd>d</kbd> and quit <kbd>q</kbd>

**Comment by millermedeiros** [comment]
one thing that I just realized after reading the help is that we don't really need a `check` command. if user passes `--chars`, `--unified` or `--unified-no-color` it's implied he wants to check the code style. both commands receive the exactly same arguments, only difference is that the `-i` flag can only be used when formatting and the diff options should be used when you want a diff..

```
$ esformatter --help
Usage: esformatter [command] [options] [files]

Commands:
  check     Check if files are formatted properly.
  format    Formats files. (default command)

Options:
  -c, --config        Path to custom configuration file.
  -v, --version       Display the current version.
  --plugins           Comma separated list of plugins.
  -i                  Edit input files in place; use with care!
  -h, --help          Display help and usage details.
  --unified           Output unified diff.
  --unified-no-color  Output unified diff without colors.
  --chars             Output char diff. (default diff format)
```

**Comment by jzaefferer** [comment]
How about prefixing the diff options with "diff", so "--diff-unified", "--diff-chars"? Or since chars is probably the best default, `--diff` (instead of `--chars`), with "--diff-unified" as an alternative.

The `no-color` option shouldn't be necessary, e.g. the `chalk` module can figure out if color is supported, afaik via https://www.npmjs.com/package/supports-color

**Comment by millermedeiros** [comment]
I'm really divided about `supports-color`... it's a neat feature but it's also kinda unintuitive, at least it supports the `--color` flag. will push a refactor that uses it soon.

**Comment by millermedeiros** [comment]
another good change was to move the cli doc to an external file: https://github.com/millermedeiros/esformatter/blob/292-check-style/doc/cli.txt - easier to update and more flexible.

File: lib/cli.js [hunk 1]
```diff
@@ -3,37 +3,43 @@
...
+
+var parseOpts = {
+  'boolean': [
+    'help',
+    'version',
+    'i',
+    'diff',
+    'diff-unified',
+    'no-color',
+    'color'
```
Comment (jzaefferer) [inline lib/cli.js]: These two don't seem to be used anywhere - leftover? If so, also drop them in `doc/cli.txt`.

File: lib/diff.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+var format = require('./format');
+var disparity = require('disparity');
+
+var hr = '==================================================================' +
+  '==============';
+
+// these headers makes more sense in this context
```
Comment (jzaefferer) [inline lib/diff.js]: typo "makes", should be "make"

**Comment by jzaefferer** [comment]
Overall looking good. Will have to figure out how to integrate all these new features into my grunt plugin. Preferably I'll delete most of it and just delegate everything to the binary.

File: lib/cli.js (hunk 1 above)
Comment (millermedeiros) [inline lib/cli.js]: `--color` and `--no-color` are used by `supports-color`

Comment (jzaefferer) [inline lib/cli.js]: Oh, okay, I didn't think of checking if that module parses argsv itself...

**Comment by millermedeiros** [comment]
squashed and landed on master. thanks for the code reviews and feedback @jzaefferer!

**Pull request closed without merging**